src/main/java/com/lambdaworks/redis/protocol/ConnectionWatchdog.java

```diff
@@ -65,7 +63,7 @@
 
     private volatile int attempts;
     private volatile boolean listenOnChannelInactive;
-    private volatile Timeout reconnectScheduleTimeout;
+    private volatile boolean isWaitingReconnection;
```

Comment:
It would make sense to represent the state (whether a reconnect is scheduled or not) via `AtomicBoolean`. It can act as lock-free synchronizer after `if ((channel == null || !channel.isActive()) && !isWaitingReconnection) {`